### PR TITLE
[v6r22] ProxyProvider resources

### PR DESCRIPTION
**Pull request by atsareg**
This PR adds a new Resource type, ProxyProvider, to encapsulate services that can generate on the fly X509 certificate proxies. The implementations include DIRAC CA and PUSP proxy providers. 

BEGINRELEASENOTES

*Resources
NEW: ProxyProvider resources with implementation for DIRAC CA and PUSP proxy providers

ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
Interesting. I guess this is just a piece of a whole development. Few questions:
- Is this used anywhere right now? (not in DIRAC, I know, but what about OAuthDIRAC?)
- is part of this code a drop-in replacement for other code in use right now?
- Should VOMS be a proxy provider?

File: Resources/ProxyProvider/ProxyProviderFactory.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+__RCSID__ = "$Id$"
+
+
+class ProxyProviderFactory(object):
+
+  #############################################################################
+  def __init__(self):
+    """ Standard constructor
+    """
+    self.log = gLogger.getSubLogger('ProxyProviderFactory')
```
Comment (andresailer) [inline Resources/ProxyProvider/ProxyProviderFactory.py]: use `__name__` for the SubLogger
(Define a `LOG = gLogger.getSubLogger(__name__)` at the top, and use that for all logging)

File: Resources/ProxyProvider/ProxyProviderFactory.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    ppDict = getProxyProviderConfigDict(proxyProvider)
+    ppType = ppDict.get('ProxyProviderType')
+    self.log.verbose('Creating ProxyProvider of %s type with the name %s' % (ppType, proxyProvider))
+    subClassName = "%sProxyProvider" % (ppType)
+
+    objectLoader = ObjectLoader.ObjectLoader()
+    result = objectLoader.loadObject('Resources.ProxyProvider.%s' % subClassName, subClassName)
+    if not result['OK']:
+      gLogger.error('Failed to load object', '%s: %s' % (subClassName, result['Message']))
```
Comment (andresailer) [inline Resources/ProxyProvider/ProxyProviderFactory.py]: Don't use glogger directly

**Comment by atsareg** [comment]
OK for merging ? Needed for OAuth... PR

File: Resources/ProxyProvider/ProxyProvider.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+""" ProxyProvider base class for various proxy providers
+"""
```
Comment (fstagni) [inline Resources/ProxyProvider/ProxyProvider.py]: Maybe you add a few lines detailing what the actual proxy providers should implement?
IIUC the only method to implement is `getProxy`

**Comment by fstagni** [comment]
IIUC this is merged in #4068 so I guess this can be closed...?

**Comment by fstagni** [comment]
Closing as merge in #4068

**Pull request closed without merging**